src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -557,7 +557,10 @@ private void ReportExportedTypeNameCollisions(ImmutableArray<Cci.ExportedType> e
                 // (type, index of the parent exported type in builder, or -1 if the type is a top-level type)
                 var stack = ArrayBuilder<(NamedTypeSymbol type, int parentIndex)>.GetInstance();
 
-                foreach (NamedTypeSymbol forwardedType in wellKnownAttributeData.ForwardedTypes)
+                // Hashset enumeration is not guarenteed to be deterministic. Emitting in the order of fully qualified names.
+                var orderedForwardedTypes = wellKnownAttributeData.ForwardedTypes.OrderBy(t => t.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
```

Comment:
>t.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat) [](start = 95, length = 59)

We probably should get display string from OriginalDefinition and use metadata format, for example "...Dictionary`2" vs. "...Dictionary<K, V>".